new-dti-website/components/team/TeamHero.tsx

```diff
@@ -1,5 +1,184 @@
-import MemberDisplay from './MemberDisplay';
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import Autoplay from 'embla-carousel-autoplay';
+import { ibm_plex_mono } from '../../src/app/layout';
+import { Carousel, CarouselContent, CarouselItem, type CarouselApi } from '../ui/carousel';
+import carouselImages from './data/carousel.json';
+import useScreenSize from '../../src/hooks/useScreenSize';
+import RedBlob from '../blob';
+import { TABLET_BREAKPOINT } from '../../src/consts';
 
-const TeamHero = () => <MemberDisplay />;
+type ImageModalProps = {
+  onClose: () => void;
+  carouselIndex: number;
+  setCarouselIndex: Dispatch<SetStateAction<number>>;
+  carouselApi: CarouselApi;
+};
+
+const ImageModal: React.FC<ImageModalProps> = ({ onClose, carouselIndex, carouselApi }) => {
+  const handleNext = () => carouselApi?.scrollNext();
+  const handlePrev = () => carouselApi?.scrollPrev();
+  return (
+    <div className="fixed w-full inset-0 bg-opacity-50 backdrop-blur-sm z-20 md:block xs:hidden">
+      <div className="flex justify-center items-center h-full gap-[70px] md:scale-75 lg:scale-100">
+        <img
+          src="/icons/arrow.svg"
+          alt="left arrow"
+          width={20}
+          onClick={handlePrev}
+          className="cursor-pointer"
+        />
+        <Carousel
+          opts={{
+            startIndex: carouselIndex,
+            loop: true
+          }}
+          className="w-[750px]"
+        >
+          <CarouselContent>
+            {carouselImages.images.map((image) => (
+              <CarouselItem key={image.alt} className="flex justify-center">
+                <div className="relative w-[750px] h-[511px] bg-white rounded-md">
+                  <div className="flex items-center h-[440px] overflow-hidden rounded-md relative top-[10px] left-[10px]">
+                    <img src={image.src} alt={image.alt} className="w-[730px] h-fit" />
+                  </div>
+                  <img
+                    src="/icons/close_icon.svg"
+                    alt="close"
+                    width={47}
+                    height={47}
+                    className="absolute right-5 top-5 cursor-pointer"
+                    onClick={onClose}
+                  />
+                  <img
+                    src={image.icon}
+                    alt="icon"
+                    width={90}
+                    height={90}
+                    className="absolute bottom-20 left-6"
+                  />
+                  <p
+                    className={`absolute bottom-0 p-5 text-xl text-[#877B7B] whitespace-nowrap ${ibm_plex_mono.className}`}
+                  >
+                    {`${carouselImages.images[carouselIndex % 6].alt}.jpg`}
```

Comment:
6 feels like a magic number to me. can we use another value to calculate 6? Or if it's temporary, can you make it a constant and move it up a level so it's easier to find/we see that it's a fixed constant?